core/server/lib/image/manipulator.js

```diff
@@ -25,18 +33,26 @@ const process = (options = {}) => {
         }));
     }
 
-    return img.metadata()
+    return init
+        .then(() => img.metadata())
         .then((metadata) => {
+            originalSize = metadata.size;
+
             if (metadata.width > options.width) {
                 img.resize(options.width);
             }
 
             // CASE: if you call `rotate` it will automatically remove the orientation (and all other meta data) and rotates
             //       based on the orientation. It does not rotate if no orientation is set.
             img.rotate();
+            return img.toBuffer({resolveWithObject: true});
         })
-        .then(() => {
-            return img.toFile(options.out);
+        .then(({data, info}) => {
+            if (info.size > originalSize) {
+                return fs.writeFile(options.out, originalData);
```

Comment:
Understanding question: we can't know if this case becomes true before manipulating the image, right?

Reply:
Yes, that's exactly the case

Reply:
Ok cool 👍